### PR TITLE
Move devDependencies to root package.json file

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Initially, I wanted to remove unused npm `devDependencies` from the root `package.json` file. In the process, it turned out that it breaks things. The issue I discovered is that local packages have only `dependencies` installed and all `devDependencies` are ignored. It seems like our setup works only because of we had all those dev dependencies installed previously from the root `package.json` file and they either are still listed there or there are dependencies of other dependencies.

To avoid the confusion in the future I removed all `devDependencies` from local packages. Instead, I moved all of them back to the root `package.json` file.

## How has this been tested?
Make sure that `npm install` still installs all dev dependencies and run all basic validations:
* `npm run dev`
* `npm run build`
* `npm test`
* `npm test-e2e`

All of that should be also covered by Travis.

## Types of changes
Refactoring

### CODE REVIEW

**Review by adamsilverstein — APPROVED** [review]
looks good to me, verified all tests passing locally.

**Comment by gziolo** [comment]
> ps. @gziolo do we need to update `package-lock.json`? After running npm install locally mine was updated.

Can you share what did change? Travis has a check for lock file changes built-in so it should error if that is the case. I often see changes in lock file which only change `http` to `https` (or the other way around) in urls.

File: packages/is-shallow-equal/package.json [hunk 1]
```diff
@@ -26,14 +26,6 @@
 	"dependencies": {
 		"@babel/runtime": "^7.0.0"
 	},
-	"devDependencies": {
-		"benchmark": "^2.1.4",
-		"fbjs": "^0.8.16",
-		"is-equal-shallow": "^0.1.3",
-		"shallow-equal": "^1.0.0",
```
Comment (aduth) [inline packages/is-shallow-equal/package.json]: Maybe these should have been made as `optionalDependencies` 😅 They're only needed for the benchmark. Caused some confusion for me looking at `package.json` changes.

**Review by aduth — APPROVED** [review]
Given that development is meant to occur only within the context of this repository, this feels very sensible.

You have a merge conflict which needs to be resolved.

File: packages/is-shallow-equal/package.json (hunk 1 above)
Comment (gziolo) [reply packages/is-shallow-equal/package.json]: From https://docs.npmjs.com/files/package.json#optionaldependencies:
> Entries in `optionalDependencies` will override entries of the same name in `dependencies`, so it’s usually best to only put in one place.

That might work, too. Benchmarking code is unfortunately very tricky in this context. You don't want to make packages published to npm to have those dependencies to be installed. On the other hand, it's confusing to keep them in the root `package.json` file. Maybe we should move benchmarks to their own private package?

Comment (aduth) [reply packages/is-shallow-equal/package.json]: I think it's such a rare use-case too that I'd be fine with them not being referenced as dependencies anywhere, and then the person running them would just install them each manually.

**Pull request merged**